core/iwasm/common/wasm_runtime_common.c

```diff
@@ -4843,19 +4845,32 @@ wasm_runtime_invoke_native(WASMExecEnv *exec_env, void *func_ptr,
             case VALUE_TYPE_F32:
                 if (n_fps < MAX_REG_FLOATS)
                     n_fps++;
+#if defined(BUILD_TARGET_RISCV32_ILP32F)
+                else if (n_ints < MAX_REG_INTS) {
+                    n_ints++;
+                }
+#endif
                 else
                     n_stacks++;
                 break;
             case VALUE_TYPE_F64:
+#if defined(BUILD_TARGET_RISCV32_ILP32) \
+    || defined(BUILD_TARGET_RISCV32_ILP32F) || defined(BUILD_TARGET_ARC)
+                if (n_ints < MAX_REG_INTS - 1) {
```

Comment:
Should we handle `if (n_fps < MAX_REG_FLOATS)` first? Like the handling for VALUE_TYPE_F32, L4845.

Reply:
Don't need it, RISCV32 ILP32F using integer register to pass double type

Reply:
OK. A little confusing for the handling or BUILD_TARGET_ARC, the original code handles `else if (n_fps == MAX_REG_FLOATS - 1) {`.

If it is an issue, how about we handle riscv, arm, arc respectively:
```C
            case VALUE_TYPE_F64:
#if defined(BUILD_TARGET_RISCV32_ILP32) \
    || defined(BUILD_TARGET_RISCV32_ILP32F)
...
#endif
if defined(BUILD_TARGET_ARM_VFP) || defined(BUILD_TARGET_THUMB_VFP)
...
#endif
if defined(BUILD_TARGET_ARC)
...
#endfi
                else {
                   ...
                }
```

Reply:
I think its OK, please refer to the origin code:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/740f499e9c2930aa202b1b7e294da9256eeba46d/core/iwasm/common/wasm_runtime_common.c#L4772-L4778

Reply:
OK, for arc, since n_fps is equivalence to n_ints and MAX_REG_INTS equals to MAX_REG_FLOATS, the below two code pieces are the same:
```C
                else if (n_fps == MAX_REG_FLOATS - 1) {
                    n_fps++;
                    n_stacks++;
                }
```
```C
                else if (n_ints == MAX_REG_INTS - 1) {
                    n_ints++;
                    n_stacks++;
                }
```